common-service-core/src/main/java/com/ca/mfaas/message/core/Message.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.mfaas.message.core;
+
+import com.ca.mfaas.message.api.ApiMessage;
+import com.ca.mfaas.message.api.ApiMessageView;
+import com.ca.mfaas.message.template.MessageTemplate;
+import org.apache.commons.lang.StringEscapeUtils;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Message creator immutable class
+ */
+public final class Message {
+
+    public static final String INVALID_KEY_MESSAGE = "com.ca.mfaas.common.invalidMessageKey";
+    public static final String INVALID_MESSAGE_TEXT_FORMAT = "com.ca.mfaas.common.invalidMessageTextFormat";
+
+    private final String requestedKey;
+    private final MessageTemplate messageTemplate;
+    private final Object[] messageParameters;
+
+
+    private Message(String requestedKey,
+                    MessageTemplate messageTemplate,
+                    Object[] messageParameters) {
+        this.requestedKey = requestedKey;
+        this.messageTemplate = messageTemplate;
+        this.messageParameters = messageParameters;
+    }
+
+    /**
+     * Return a {@link Message} object for the specified key after text and parameters validation.
+     *
+     * @param requestedKey the message key.
+     * @param messageTemplate the messageTemplate.
+     * @param messageParameters the object containing the message parameters.
+     * @return {@link Message}
+     */
+    public static Message of(String requestedKey,
+                             MessageTemplate messageTemplate,
+                             Object[] messageParameters) {
+
+        requireNotNull(requestedKey, "requestedKey can't be null");
+        requireNotNull(messageTemplate, "messageTemplate can't be null");
+        requireNotNull(messageParameters, "messageParameters can't be null");
+
+        validateMessageTextFormat(messageTemplate.getText(), messageParameters);
+        messageParameters = validateParameters(messageTemplate.getKey(), requestedKey, messageParameters);
+
+        return new Message(requestedKey, messageTemplate, messageParameters);
+    }
+
+
+    /**
+     * Checks that the specified object reference is not null and
+     * throws a {@link IllegalArgumentException} if it is.
+     *
+     * @param param the object reference to check for nullity
+     * @param message detail message to be used in the event
+     */
+    private static void requireNotNull(Object param, String message) {
+        if (param == null) {
+            throw new  IllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Validate the message text and parameters returning them as formatted String.
```

Comment:
What do you require from a user? To validate text in the message or to validate the parameters of the message? In the case of parameters, what parameters need to be validated? Now, ".... returning them as formatted String", parameters of what do you refer to? Parameters of the message or parameters inside the message? I know, too many questions.